source/globalCommands.py

```diff
@@ -2278,6 +2282,49 @@ def script_recognizeWithUwpOcr(self, gesture):
 	# Translators: Describes a command.
 	script_recognizeWithUwpOcr.__doc__ = _("Recognizes the content of the current navigator object with Windows 10 OCR")
 
+	@script(
+		# Translators: Describes a command.
+		description=_(
+			"Toggles the state of the screen curtain, "
+			"either by making the screen black or SHOWING the contents of the screen. "
+			"If pressed to enable once, the screen curtain is enabled until you restart NVDA. "
+			"If pressed tree times, it is enabled until you disable it"
+		),
+		category=SCRCAT_VISION
+	)
+	def script_toggleScreenCurtain(self, gesture):
+		message = None
+		try:
+			screenCurtainName = "screenCurtain"
+			if not vision.getProviderClass(screenCurtainName).canStart():
+				# Translators: Reported when the screen curtain is not available.
+				message = _("Screen curtain not available")
+				return
+			scriptCount = scriptHandler.getLastScriptRepeatCount()
+			if scriptCount == 0 and screenCurtainName in vision.handler.providers:
+				vision.handler.terminateProvider(screenCurtainName)
+				# Translators: Reported when the screen curtain is disabled.
+				message = _("Screen curtain disabled")
+			elif scriptCount in (0, 2):
+				temporary = scriptCount == 0
+				if not vision.handler.initializeProvider(
+					screenCurtainName,
+					temporary=temporary,
+				):
+					# Translators: Reported when the screen curtain could not be enabled.
+					message = _("Could not enable screen curtain")
+					return
+				else:
+					if temporary:
+						# Translators: Reported when the screen curtain is temporarily enabled.
+						message = _("Screen curtain enabled until next restart")

```

Comment:
When pressing this multiple times, it might be better to hear the "differentiation" between each `scriptCount` outcome first. However I currently can't think of a nice way to word it. Maybe "Temporary Screen Curtain, enabled until next restart". It's longer, but tells you the important part first, power users can skip the rest.

Reply:
This makes sense. Applied as suggested.